server/src/test/java/org/elasticsearch/index/mapper/BinaryRangeUtilTests.java

```diff
@@ -140,6 +146,61 @@ public void testEncode_Float() {
         }
     }
 
+    public void testDecodeLong() {
+        long[] cases = new long[] { Long.MIN_VALUE, -2049, -2048, -128, -3, -1, 0, 1, 3, 125, 2048, 2049, Long.MAX_VALUE};
+        for (long expected : cases) {
+            byte[] encoded = BinaryRangeUtil.encodeLong(expected);
+            int offset = 0;
+            int length = BinaryRangeUtil.LengthType.VARIABLE.readLength(encoded, offset);
+            assertEquals(expected, BinaryRangeUtil.decodeLong(encoded, offset,  length));
+        }
+    }
+
+    public void testDecodeLongRanges() throws IOException {
+        long start = randomLong();
+        long end = randomLongBetween(start + 1, Long.MAX_VALUE);
+        RangeFieldMapper.Range expected = new RangeFieldMapper.Range(RangeFieldMapper.RangeType.LONG, start, end, true, true);
+        List<RangeFieldMapper.Range> decoded = BinaryRangeUtil.decodeLongRanges(BinaryRangeUtil.encodeLongRanges(singleton(expected)));
+        assertEquals(1, decoded.size());
+        RangeFieldMapper.Range actual = decoded.get(0);
+        assertEquals(expected, actual);
+    }
+
+    public void testDecodeDoubleRanges() throws IOException {
```

Comment:
maybe test more than one random value, a few iteration should be enough ?

Reply:
done.